controllers/spacecleanup/space_cleanup_controller.go

```diff
@@ -92,6 +92,11 @@ func (r *Reconciler) ensureDeletionIfNeeded(logger logr.Logger, space *toolchain
 		return false, 0, nil
 	}
 
+	// check if space has an "active" parentSpace
+	if exists, err := r.parentSpaceExists(logger, space); exists {
```

Comment:
what happens if `!exists && err != nil` ?

Reply:
Thanks @xcoulon ! I've added a return of the err even in that case! 

See: [7eda771](https://github.com/codeready-toolchain/host-operator/pull/728/commits/7eda771d353d09901d0c50f121026a686e761946)